### PR TITLE
Refactor and redesign welcome screen

### PR DESCRIPTION
**Pull request by XavierJp**

### CODE REVIEW

File: src/atoms/buttonLink.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        type={type}
+        alt={alt}
+        disabled={disabled}
+        className={className}
+        children={children}
+        size={size}
+      />
+    );
+  }
+  throw new Error('Should not happen');
```
Comment (rdubigny) [inline src/atoms/buttonLink.js:84]: message could be : « Please specify either 'href', 'onClick' or 'type' props »

File: src/atoms/buttonLink.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        alt={alt}
+        disabled={disabled}
+        children={children}
+        size={size}
+        onClick={onClick}
+        className={className}
+        referrerPolicy={referrerPolicy}
+      />
+    );
+  }
```
Comment (rdubigny) [inline src/atoms/buttonLink.js:70]: you could make a great use of spread operator like so : 

```suggestion
const ButtonLink = ({
  href,
  type,
  onClick,
  ...props,
}) => {
  if (href) {
    return (
      <LinkAsAButton
        href={href}
        {...props}
      />
    );
  }
```

this way, future addition of props should be easier

File: src/atoms/buttonLink.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
```
Comment (rdubigny) [inline src/atoms/buttonLink.js:1]: yet another implicit convention for name of file that export react component : use capital letter like so "**B**uttonLink.js"

Comment (rdubigny) [inline src/atoms/buttonLink.js:1]: also, the atom folder should be a subfolder of `src/components`

File: src/components/icons/auth.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const auth = ({ color }) => (
```
Comment (rdubigny) [inline src/components/icons/auth.js:4]: convention : auth => AuthIcon

File: src/components/icons/demande.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const demande = ({ color }) => (
```
Comment (rdubigny) [inline src/components/icons/demande.js:4]: convention : demande => DemandeIcon

File: src/components/icons/habilitation.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const habilitation = ({ color = undefined }) => (
```
Comment (rdubigny) [inline src/components/icons/habilitation.js:4]: convention

File: src/components/icons/token.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const token = ({ color }) => (
```
Comment (rdubigny) [inline src/components/icons/token.js:4]: convention

File: src/pages/Login/WelcomeMessage.js [hunk 8]
```diff
@@ -1,33 +1,37 @@
...
-
-  if (TARGET_API_LABELS[targetApi])
-    return `Vous souhaitez suivre le traitement d'une demande d'habilitation à « ${
-      TARGET_API_LABELS[targetApi]
-    } », `;
-
-  return "Vous souhaitez suivre le traitement d'une demande d'habilitation, ";
+const getWelcomeMessage = ({ isOnNewEnrollmentPage, targetApi }) => {
+  let message = '';
+  if (isOnNewEnrollmentPage) {
```
Comment (rdubigny) [inline src/pages/Login/WelcomeMessage.js:8]: as a fervent « [guard clause](https://refactoring.guru/replace-nested-conditional-with-guard-clauses) » supporter, I am not found of this rewriting. This new code is more prone to be extended into a nested conditional antipattern

File: src/pages/Login/WelcomeMessage.js [hunk 9]
```diff
@@ -45,14 +49,33 @@ const WelcomeMessage = ({ isOnNewEnrollmentPage, targetApi }) => {
...
+          </p>
         </>
       )}
-      <p className="call-to-action">
-        {welcomeMessage}
-        {isOnNewEnrollmentPage
-          ? 'merci de créer un compte pour déposer votre demande et suivre son traitement.'
-          : 'merci de vous identifier afin que nous puissions configurer vos accès.'}
-      </p>
+      <div>
```
Comment (rdubigny) [inline src/pages/Login/WelcomeMessage.js:59]: this div seems of no use !

File: src/pages/Login/index.css [hunk 10]
```diff
@@ -18,3 +18,69 @@
...
 }
+
+.next-steps,
+.timeline {
+  display: flex;
+  justify-content: space-between;
+  align-items: start;
+  margin: 30px 0 15px;
+}
+.timeline {
```
Comment (rdubigny) [inline src/pages/Login/index.css:29]:
```suggestion
}

.timeline {
```

File: src/pages/Login/index.css [hunk 11]
```diff
@@ -18,3 +18,69 @@
...
+.timeline > div {
+  width: 24%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  color: var(--blue);
+  font-weight: bold;
+}
+.timeline > div {
```
Comment (rdubigny) [inline src/pages/Login/index.css:44]:
```suggestion
}

.timeline > div {
```

File: src/pages/Login/index.css [hunk 12]
```diff
@@ -18,3 +18,69 @@
...
+
+.timeline > div:before,
+.timeline > div:after {
+  content: '';
+  height: 3px;
+  width: 50%;
+  background-color: var(--blue);
+  opacity: 0.15;
+}
+.timeline > div:first-of-type:before,
```
Comment (rdubigny) [inline src/pages/Login/index.css:69]:
```suggestion
}

.timeline > div:first-of-type:before,
```

File: src/pages/Login/index.css [hunk 13]
```diff
@@ -18,3 +18,69 @@
...
+.timeline > div:last-of-type:after {
+  opacity: 0;
+}
+.timeline .step {
```
Comment (rdubigny) [inline src/pages/Login/index.css:73]:
```suggestion
}

.timeline .step {
```

File: src/pages/Login/index.css [hunk 14]
```diff
@@ -18,3 +18,69 @@
...
+  height: 7px;
+  width: 7px;
+  border-radius: 15px;
+  background-color: var(--blue);
+}
+
+.next-steps svg {
+  fill: var(--blue);
+}
+.next-steps .prepare > ul {
```
Comment (rdubigny) [inline src/pages/Login/index.css:83]:
```suggestion
}

.next-steps .prepare > ul {
```

File: src/pages/Login/index.css [hunk 15]
```diff
@@ -18,3 +18,69 @@
 .call-to-action {
   margin-bottom: 3em;
 }
+
+.next-steps,
+.timeline {
+  display: flex;
+  justify-content: space-between;
+  align-items: start;
```
Comment (rdubigny) [inline src/pages/Login/index.css:26]: this trigger a warning : (26:3) start value has mixed support, consider using flex-start instead

File: src/pages/Login/index.js [hunk 16]
```diff
@@ -5,10 +5,35 @@ import { API_ICONS, TARGET_API_LABELS } from '../../lib/api';
 import WelcomeMessage from './WelcomeMessage';
 import ApiImpotParticulierFcSandboxWelcomeMessage from './ApiImpotParticulierFcSandboxWelcomeMessage';
 import ApiImpotParticulierSandboxWelcomeMessage from './ApiImpotParticulierSandboxWelcomeMessage';
+import ButtonLink from '../../atoms/buttonLink';
 
 const { REACT_APP_BACK_HOST: BACK_HOST } = process.env;
 
-export const Login = () => {
+const WelcomeMessageRouter = ({ targetApi, isOnNewEnrollmentPage }) => {
```
Comment (rdubigny) [inline src/pages/Login/index.js:12]: :heart:

File: src/pages/Login/WelcomeMessage.js [hunk 17]
```diff
@@ -45,14 +49,33 @@ const WelcomeMessage = ({ isOnNewEnrollmentPage, targetApi }) => {
               En savoir plus
             </a>
           </p>
+          <p>
+            Pour faire une demande d'accès, allez sur{' '}
+            <a href="https://api.gouv.fr/rechercher-api">api.gouv</a> et faites
+            une demande à l'API de votre choix.
+          </p>
```
Comment (rdubigny) [inline src/pages/Login/WelcomeMessage.js:56]: I do not get what problem is solved by this. To me this seems not useful in most of the cases.

case 1 : user coming from api.gouv.fr might click on this and make a loop that push him back where he came from.
case 2 : user coming from email notification (ex : « your demand as been validated », or « you are the dpd contact »), might not find this link useful.
case 3 : people coming directly to https://datapass.api.gouv.fr might find this link useful but this kind of access must be from instructors mostly.

I suggest to rather improve or replace the previous <p> to keep this page simple.

**Review by rdubigny — CHANGES_REQUESTED** [review]
Thanks for this beautiful hompage. I'am eager to see even more simplification of this messy code (my bad) !

I will squash the commits. In the future, could you rebase your commits to avoid commit like 'remove package_lock.json'.

**Comment by rdubigny** [comment]
[Implémenter l'écran d'acceuil](https://trello.com/c/2BHd4K9C/816-impl%C3%A9menter-l%C3%A9cran-dacceuil)

File: src/atoms/buttonLink.js (hunk 1 above)
Comment (XavierJp) [reply src/atoms/buttonLink.js:84]: ok

File: src/atoms/buttonLink.js (hunk 2 above)
Comment (XavierJp) [reply src/atoms/buttonLink.js:70]: plus opaque en l'absence de typescript mais ok

File: src/atoms/buttonLink.js (hunk 3 above)
Comment (XavierJp) [reply src/atoms/buttonLink.js:1]: ok

Comment (XavierJp) [reply src/atoms/buttonLink.js:1]: ok pour moi c'est différent ca les component ont une logique metiers, alors que les atoms sont purement fonctionnels

File: src/components/icons/auth.js (hunk 4 above)
Comment (XavierJp) [reply src/components/icons/auth.js:4]: ok je fais ca pour les 4 alors

File: src/pages/Login/index.css (hunk 13 above)
Comment (XavierJp) [reply src/pages/Login/index.css:73]: Tu as des warnings en local, ou prettier ? Je n'ai rien 🤔 ?

File: src/pages/Login/index.css (hunk 14 above)
Comment (XavierJp) [reply src/pages/Login/index.css:83]: same

File: src/pages/Login/index.css (hunk 15 above)
Comment (XavierJp) [reply src/pages/Login/index.css:26]: same

File: src/pages/Login/WelcomeMessage.js (hunk 17 above)
Comment (XavierJp) [reply src/pages/Login/WelcomeMessage.js:56]: Plutôt pas d'accord mais ok pour supprimer comme ca on déploie

**Review by rdubigny — APPROVED** [review]

**Pull request merged**